Tribler/Test/Core/CreditMining/test_creditmining.py

```diff
@@ -0,0 +1,252 @@
+# coding=utf-8
+"""
+Module of Credit mining function testing
+
+Written by Mihai Capotă and Ardhi Putra Pratama H
+"""
+
+import random
+
+import re
+
+import Tribler.Policies.BoostingManager as bm
+from Tribler.Core.Libtorrent.LibtorrentDownloadImpl import LibtorrentDownloadImpl
+from Tribler.Core.Utilities import utilities
+from Tribler.Policies.BoostingSource import ent2chr
+from Tribler.Test.test_as_server import TestAsServer
+
+
+class MockMeta(object):
+    """
+    class for mocking the torrent metainfo
+    """
+
+    def __init__(self, id_hash):
+        self.infohash = id_hash
+
+    def get_infohash(self):
+        """
+        returning infohash of torrents
+        """
+        return self.infohash
+
+
+class TestBoostingManagerPolicies(TestAsServer):
```

Comment:
As a general rule, please try to not boot up a complete session if it's not necessary. I think most, if not all, methods in this class can be unit tested with some basic mocking of the session :)